From 8afd91a52f86d7fd00c44df1cf8e4d73dab20b63 Mon Sep 17 00:00:00 2001
From: Alisher Alikhodjaev <alisher@google.com>
Date: Tue, 19 Oct 2021 16:36:26 -0700
Subject: [PATCH 2/2] [Q_asb_2022-01] Use after free in phTmlNfc_TmlThread

Bug: 197353344
Test: build ok
Merged-In: Id6323fd3f0588972dba3b7654435c171d6f5279b
Change-Id: I1f318ae8f0263177040e10baa25952c1accdea21
(cherry picked from commit 647f3fcc75511c6f05e1670d2457b0d93c13abdf)
Merged-In: I1f318ae8f0263177040e10baa25952c1accdea21
---
 halimpl/tml/phTmlNfc.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/halimpl/tml/phTmlNfc.cc b/halimpl/tml/phTmlNfc.cc
index 5d33b23..042a345 100755
--- a/halimpl/tml/phTmlNfc.cc
+++ b/halimpl/tml/phTmlNfc.cc
@@ -151,7 +151,7 @@ NFCSTATUS phTmlNfc_Init(pphTmlNfc_Config_t pConfig) {
   /* Clean up all the TML resources if any error */
   if (NFCSTATUS_SUCCESS != wInitStatus) {
     /* Clear all handles and memory locations initialized during init */
-    phTmlNfc_CleanUp();
+    phTmlNfc_Shutdown_CleanUp();
   }
 
   return wInitStatus;
-- 
2.35.1

